feature-libs/order/components/order-details/order-details-extended/consignment-tracking/consignment-tracking-link.component.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  inject,
+  OnDestroy,
+  OnInit,
+  ViewChild,
+  ViewContainerRef,
+} from '@angular/core';
+import {
+  LaunchDialogService,
+  LAUNCH_CALLER,
+  OutletContextData,
+} from '@spartacus/storefront';
+import { of, Subscription } from 'rxjs';
+import { take } from 'rxjs/operators';
+import { Consignment } from '@spartacus/order/root';
+import { OrderConsignmentsService } from '../../order-consignments.service';
+
+@Component({
+  selector: 'cx-consignment-tracking-link',
+  templateUrl: './consignment-tracking-link.component.html',
+  host: { class: 'cx-list-header col-12' },
+})
+export class ConsignmentTrackingLinkComponent implements OnInit, OnDestroy {
+  consignment: Consignment;
+  protected subscription = new Subscription();
+  protected orderConsignmentsService = inject(OrderConsignmentsService);
+  protected launchDialogService = inject(LaunchDialogService);
+  protected vcr = inject(ViewContainerRef);
+  protected cd = inject(ChangeDetectorRef);
+  protected outlet = inject(OutletContextData);
+
+  @ViewChild('element') element: ElementRef;
+  consignmentStatus: string[] = this.orderConsignmentsService.consignmentStatus;
+  ngOnInit(): void {
+    this.subscription.add(
+      this.outlet?.context$.subscribe((context: { item?: Consignment }) => {
+        if (context.item !== undefined) {
+          this.consignment = context.item;
+        }
+        this.cd.markForCheck();
+      })
```

Comment:
[MEDIUM]
I don't recommend mixing observable programming with imperative programming (e.g. subscribing to observables and copying values to the component's static properties), unless _really_ justified.

It's prone to issues with synchronization of data and the change detection issues (and often requires calling `cd.markForCheck()` manually). Moreover it requires manually creating and destroying a subscription.

Instead, I'd suggest just using the observables as far as possible (e.g. pass observable to in the template and unwrap the value there with `async` pipe)

In this example, I suggest to simply create an observable that derives from the `outlet.context$` observable via `map()` operator:
```ts
this.consignment$ = this.outlet?.context$.pipe(
  map(context => context.item)
);
```
then unwrap this observable in the html template:
```html
<span *ngIf="consignment$ | async as consignment">
   ...
   <button (click)="openTrackingDialog(consignment)> <!-- NOTE: passing object to method ! -->
```
then add parameter to the TS `openTrackingDialog()` method:
```ts
  openTrackingDialog(consignment: WhateverTypeAppropriate_Perhaps_Consignment) {
    ...
  }
```

--- 

Note: the analogical remark applies also to other files in this PR. Please review other subscriptions in components in this PR, and check if we could simply create a derived observable instead of copying values from the source observable to a static variable.

Reply:
done